solr/core/src/test/org/apache/solr/handler/admin/CoreAdminHandlerActionTest.java

```diff
@@ -0,0 +1,157 @@
+import org.apache.solr.common.util.SimpleOrderedMap;
+import org.apache.solr.handler.admin.CoreAdminHandler.CallInfo;
+import org.apache.solr.handler.admin.CoreAdminHandler.CoreAdminOp;
+import org.apache.solr.response.SolrQueryResponse;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class CoreAdminHandlerActionTest extends SolrTestCaseJ4 {
+
+  private static CoreAdminHandler admin = null;
+
+  @BeforeClass
+  public static void beforeClass() throws Exception {
+
+    setupNoCoreTest(createTempDir(), null);
+
+    admin = new CoreAdminHandler(h.getCoreContainer());
+
+    admin.registerCustomActions(
+        Map.of(
+            "action1",
+            new CoreAdminHandlerTestAction1(),
+            "action2",
+            new CoreAdminHandlerTestAction2()));
+  }
+
+  @Test
+  @SuppressWarnings("unchecked")
+  public void testAction1() throws Exception {
+
+    SolrQueryResponse response = new SolrQueryResponse();
+
+    admin.handleRequestBody(req(CoreAdminParams.ACTION, "action1"), response);
+
+    Map<String, Object> actionResponse = ((NamedList<Object>) response.getResponse()).asMap();
+
+    assertTrue(
+        String.format(
+            "Action response should contain %s property",
+            CoreAdminHandlerTestAction1.PROPERTY_NAME),
+        actionResponse.containsKey(CoreAdminHandlerTestAction1.PROPERTY_NAME));
+    assertEquals(
+        String.format(
+            "Action response should contain %s value for %s property",
+            CoreAdminHandlerTestAction1.PROPERTY_VALUE, CoreAdminHandlerTestAction1.PROPERTY_NAME),
+        CoreAdminHandlerTestAction1.PROPERTY_VALUE,
+        actionResponse.get(CoreAdminHandlerTestAction1.PROPERTY_NAME));
+  }
+
+  @Test
+  @SuppressWarnings("unchecked")
+  public void testAction2() throws Exception {
+
+    SolrQueryResponse response = new SolrQueryResponse();
+
+    admin.handleRequestBody(req(CoreAdminParams.ACTION, "action2"), response);
+
+    Map<String, Object> actionResponse = ((NamedList<Object>) response.getResponse()).asMap();
+
+    assertTrue(
+        String.format(
+            "Action response should contain %s property",
+            CoreAdminHandlerTestAction2.PROPERTY_NAME),
+        actionResponse.containsKey(CoreAdminHandlerTestAction2.PROPERTY_NAME));
+    assertEquals(
+        String.format(
+            "Action response should contain %s value for %s property",
+            CoreAdminHandlerTestAction2.PROPERTY_VALUE, CoreAdminHandlerTestAction2.PROPERTY_NAME),
+        CoreAdminHandlerTestAction2.PROPERTY_VALUE,
+        actionResponse.get(CoreAdminHandlerTestAction2.PROPERTY_NAME));
+  }
+
+  @Test
+  public void testUnregisteredAction() throws Exception {
+
+    SolrQueryResponse response = new SolrQueryResponse();
+
+    assertThrows(
+        "Attempt to execute unregistered action should throw SolrException",
+        SolrException.class,
+        () -> admin.handleRequestBody(req(CoreAdminParams.ACTION, "action3"), response));
+  }
```

Comment:
Maybe there is a similar method already implemented in some utils class I can use?